src/parser.cr

```diff
@@ -93,6 +90,14 @@ module Mint
     # Consuming characters
     # ----------------------------------------------------------------------------
 
+    def letters_or_numbers
+      chars { |char| char.ascii_letter? || char.ascii_number? }
+    end
+
+    def letters_or_numbers_or_dash
+      chars { |char| char.ascii_letter? || char.ascii_number? || char == '-' }
+    end
```

Comment:
I added these because they are used more than once. Feel free to suggest other names! Or we can inline that logic in the caller. But maybe it's good to have these named methods, it might also make the code slightly clearer.

Reply:
Done!